packages/dev/core/src/Engines/WebGPU/Extensions/engine.computeShader.ts

```diff
@@ -67,6 +77,29 @@ WebGPUEngine.prototype.computeDispatch = function (
     z = 1,
     bindingsMapping?: ComputeBindingMapping,
     gpuPerfCounter?: WebGPUPerfCounter
+): void {
+    this._computeDispatch(effect, context, bindings, { x, y, z }, bindingsMapping, gpuPerfCounter);
+};
+
+WebGPUEngine.prototype.computeDispatchIndirect = function (
+    effect: ComputeEffect,
+    context: IComputeContext,
+    bindings: ComputeBindingList,
+    buffer: DataBuffer,
+    offset: number = 0,
+    bindingsMapping?: ComputeBindingMapping,
+    gpuPerfCounter?: WebGPUPerfCounter
+): void {
+    this._computeDispatch(effect, context, bindings, { buffer, offset }, bindingsMapping, gpuPerfCounter);
+};
+
+WebGPUEngine.prototype._computeDispatch = function (

```

Comment:
As this is an internal function, I think we can declare it like:
```typescript
WebGPUEngine.prototype._computeDispatch = function (
    effect: ComputeEffect,
    context: IComputeContext,
    bindings: ComputeBindingList,
    x?: number,
    y?: number,
    z?: number,    
    buffer?: DataBuffer,
    offset?: number = 0)
```
and consider it an indirect dispatch if `buffer` is not empty, and a direct dispatch otherwise. This avoids the need to create new objects to pass parameters.

Reply:
Sounds reasonable. It's moments like these where I wish Javascript had [structs](https://github.com/tc39/proposal-structs).